OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -40,24 +40,37 @@ public class McvManagerBotModuleInfo : ConditionalTraitInfo
 		public readonly int MinBaseRadius = 2;
 
 		[Desc("Maximum distance in cells from center of the base when checking for MCV deployment location.",
-			"Only applies if RestrictMCVDeploymentFallbackToBase is enabled and there's at least one construction yard.")]
+			"AI will not send MCVs to locations containing enemies within the MaxBaseRadius of that location.",
+			"AI will not send MCVs to locations containing its own MCVs or construction yards within the MaxBaseRadius of that location.",
+			"In the case of less than 1 construction yard, the AI will choose a random new base center and place the MCV within the MaxBaseRadius of that location.",
+			"In the case of at least 1 construction yard, the AI will choose a location nearer resource fields outside of all bases MaxBaseRadius up to world.Map.Grid.MaximumTileSearchRange.")]
 		public readonly int MaxBaseRadius = 20;
 
-		[Desc("Should deployment of additional MCVs be restricted to MaxBaseRadius if explicit deploy locations are missing or occupied?")]
-		public readonly bool RestrictMCVDeploymentFallbackToBase = true;
-
 		public override object Create(ActorInitializer init) { return new McvManagerBotModule(init.Self, this); }
 	}
 
 	public class McvManagerBotModule : ConditionalTrait<McvManagerBotModuleInfo>, IBotTick, IBotPositionsUpdated, IGameSaveTraitData
 	{
-		public CPos GetRandomBaseCenter()
+		public CPos GetRandomBaseCenter(bool distanceToBaseIsImportant)
 		{
+			if (!distanceToBaseIsImportant)
+				return initialBaseCenter;
+
+			var tileset = world.Map.Rules.TileSet;
```

Comment:
I suspect it'd be good to store this in an instance variable upon construction so we aren't pointer-chasing each time we attempt to determine a base center.